### PR TITLE
add probes now that expose paths are supported

### PR DESCRIPTION
**Pull request by ndhanushkodi**
Changes proposed in this PR:
- Add http probes to acceptance test
- Made fixes in consul here: https://github.com/hashicorp/consul/pull/19257 

How I've tested this PR:

How I expect reviewers to test this PR:

Checklist:
- [ ] Tests added
- [ ] [CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

**Review by DanStough — APPROVED** [review]
Probably a little sus we don't need to make any test changes but the change looks good for now.

**Comment by ndhanushkodi** [comment]
@DanStough oh yeah, so before making fixes in consul, it the multiport pod wouldn't come up with the probes, so I saw it crashloop, and the fact that it came up and the rest of the test could run means probes are working

**Comment by ndhanushkodi** [comment]
@DanStough ahhh got it

**Review by thisisnotashwin — APPROVED** [review]

File: acceptance/tests/mesh_v2/mesh_inject_test.go [hunk 1]
```diff
@@ -35,6 +35,7 @@ func TestMeshInject_MultiportService(t *testing.T) {
 
 			helmValues := map[string]string{
 				"global.experiments[0]": "resource-apis",
+				"global.image":          "ndhanushkodi/consul-dev:expose2",
```
[the remark this answers is not in the document]
Comment (zalimeni) [reply acceptance/tests/mesh_v2/mesh_inject_test.go:38]: Just to save you the time checking: `hashicorppreview/consul:1.17.0-dev` was published this AM, so I think you're good!

**Pull request merged**